packages/interface/src/components/admin-menu-toggle/index.js

```diff
@@ -0,0 +1,198 @@
+/**
+ * WordPress dependencies
+ */
+import { Button } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { useRef, useEffect, useState } from '@wordpress/element';
+import { ESCAPE, TAB } from '@wordpress/keycodes';
+import { addQueryArgs } from '@wordpress/url';
+import { __ } from '@wordpress/i18n';
+import { wordpress } from '@wordpress/icons';
+
+function AdminMenuToggle() {
+	const href = useAdminUrl();
+	const buttonRef = useRef();
+	const { isOpen, toggleMenu } = useToggle( { ref: buttonRef } );
+
+	const handleOnClick = ( event ) => {
+		const { button, ctrlKey, metaKey } = event;
+		const isMiddleClick = button === 2;
+
+		/**
+		 * Enables the ability to open the link in a new tab by middle clicking,
+		 * or clicking while holding Command (Mac) or Control.
+		 */
+		if ( isMiddleClick || ctrlKey || metaKey ) {
+			return;
+		}
+
+		/**
+		 * Otherwise, toggle the admin menu.
+		 */
+		event.preventDefault();
+		toggleMenu();
+	};
+
+	const label = isOpen
+		? __( 'Hide sidebar menu' )
```

Comment:
Technically this is a "sidebar menu", but isn't this the "admin navigation" of a users-site?
"Hide navigation" or "Hide admin navigation" would make more sense to me as a (non-technical) user.

Reply:
Thanks for the feedback! I'll make that update